tests/smoke/support/page/files/allFiles.ts

```diff
@@ -187,4 +213,132 @@ export class AllFilesPage {
     })
     await page.goto(startUrl)
   }
+
+  async resourceExist({ name }: { name: string }): Promise<boolean> {
+    const { page } = this.actor
+    const folderPaths = name.split('/')
+    const resouceName = folderPaths.pop()
+
+    if (folderPaths.length) {
+      await cta.files.navigateToFolder({ page: page, path: folderPaths.join('/') })
+    }
+
+    const resourceExists = await cta.files.resourceExists({
+      page: page,
+      name: resouceName
+    })
+
+    return resourceExists
+  }
+
+  async numberOfVersions({ resource }: { resource: string }): Promise<number> {
```

Comment:
not sure about that, i mean it's ok to have it. I just don't think that this strictly belongs to the allFiles page only... what is we wan't to check the versions as a public share receiver?

Reply:
please, just not in this monster PR. let me do it in a separate task and spread it over by pages. we only have 2 Pages, and the Page Object pattern is broken in many places

Reply:
fine for me, can you please keep track of it :) Thanks